test/output-component-factory-mocha.js

```diff
@@ -0,0 +1,111 @@
+// output-component-factory-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var componentFactory = require('../components/output-component-factory');
+
+describe('output-component-factory', function() {
+    it("should accept array of inPort names", function() {
+        return Promise.resolve({
+            inPorts:['input'],
+            onchange:function(){}
+        }).then(componentFactory).then(createComponent).then(function(component){
+            return _.keys(component.inPorts);
+        }).should.eventually.contain('input');
+    });
+    it("should resolve onchange function", function() {
+        return Promise.resolve({
+            inPorts:['input'],
+            onchange: function(payloads) {
+                return payloads.input + " world";
+            }
+        }).then(componentFactory).then(createComponent).then(function(component){
+            return onceData(component, 'input', "hello");
+        }).should.become("hello world");
+    });
+    it("should send multiple packets on array result for onchange function", function() {
+        return Promise.resolve({
+            inPorts:['input'],
+            onchange: function(payloads) {
+                return [payloads.input, payloads.input + " world"];
+            }
+        }).then(componentFactory).then(createComponent).then(function(component){
+            return new Promise(function(resolve) {
+                var output = noflo.internalSocket.createSocket();
+                component.outPorts.output.attach(output);
+                output.once('data', function(){
+                    output.once('data', resolve);
+                });
+                sendData(component, 'input', "hello");
+            });
+        }).should.become("hello world");
+    });
+    it("should reject onchange function", function() {
+        return Promise.resolve({
+            inPorts:['input'],
+            onchange: function(payloads) {
+                throw payloads.input + " world";
+            }
+        }).then(componentFactory).then(createComponent).then(function(component){
+            return onceData(component, 'input', "hello");
+        }).should.be.rejectedWith("hello world");
+    });
+    it("should log rejections to console.error when no reject port attached", function() {
+        return Promise.resolve({
+            inPorts:['input'],
+            onchange: function(payloads) {
+                throw payloads.input + " world";
+            }
+        }).then(componentFactory).then(createComponent).then(function(component){
+            var console_error = console.error;
+            return new Promise(function(resolve, reject) {
+                console.error = reject;
+                var output = noflo.internalSocket.createSocket();
```

Comment:
Can we make this block (send data on output port once) into a common helper function since it is used a fair bit?